Sasquatch/Sasquatch/ViewControllers/MSStorageViewController.swift

```diff
@@ -3,102 +3,253 @@
+    if self.storageType == StorageType.App.rawValue {
+      return MSStorageViewController.AppDocuments.count
+    } else if self.storageType == StorageType.User.rawValue {
+      if identitySignIn {
+        return MSStorageViewController.UserDocuments.count + 1
+      } else {
+        return 0
+      }
+    }
+    return 1
+  }
+  
+  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    let cellIdentifier = "document"
     let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath)
-    if indexPath.section == 0 {
-      let backButton: UIButton? = cell.getSubview()
-      backButton?.addTarget(self, action: #selector(backButtonClicked), for: .touchUpInside)
-    } else if indexPath.section == 1 {
-      let storageTypeField: UITextField? = cell.getSubview()
-      self.storageTypePicker = MSEnumPicker<StorageType> (
-        textField: storageTypeField,
-        allValues: StorageType.allValues,
-        onChange: { index in
-          self.storageType = (storageTypeField?.text)!
-          self.tableView.reloadSections([2], with: .none)
-        }
-      )
-      storageTypeField?.delegate = self.storageTypePicker
-      storageTypeField?.tintColor = UIColor.clear
-    } else if indexPath.section == 2 {
-      cell.accessoryType = .disclosureIndicator
-      if (self.storageType == StorageType.App.rawValue) {
-        cell.textLabel?.text = AppDocuments[indexPath.row]
-      } else if (self.storageType == StorageType.User.rawValue) {
-        cell.textLabel?.text = UserDocuments[indexPath.row]
+    if self.storageType == StorageType.App.rawValue {
+      cell.textLabel?.text = MSStorageViewController.AppDocuments[indexPath.row].documentId
+    } else if self.storageType == StorageType.User.rawValue {
+      if indexPath.row == 0 {
+        cell.textLabel?.text = "Add document"
+      } else {
+        cell.textLabel?.text = MSStorageViewController.UserDocuments[indexPath.row - 1].documentId
       }
     }
     return cell
   }
-
-  override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+  
+  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
     tableView.deselectRow(at: indexPath, animated: true)
-    let cell = tableView.cellForRow(at: indexPath)
-    if indexPath.section == 2 {
-      let documentTitle = cell?.textLabel?.text
-      self.performSegue(withIdentifier: "ShowDocumentDetails", sender: documentTitle)
+    if isInsertRow(indexPath) {
+      self.performSegue(withIdentifier: "ShowDocumentDetails", sender: "")
+    } else {
+      if self.storageType == StorageType.App.rawValue {
+        self.performSegue(withIdentifier: "ShowDocumentDetails", sender: MSStorageViewController.AppDocuments[indexPath.row])
+      } else {
+        self.performSegue(withIdentifier: "ShowDocumentDetails", sender: MSStorageViewController.UserDocuments[indexPath.row - 1])
```

Comment:
I could reproduce a crash here caused by an `index out of range`. Easy to reproduce: login via Identity, go to storage, select User "storage type", (List of docs must be empty) Add  a document, (There is now only 1 doc) delete that document, try add another > crash.